server/routes/profile.js

```diff
@@ -39,6 +39,10 @@ router.put("/profile/:id", auth, upload.single("profile"), async (req, res) => {
     return res.status(404).send("Profile not found.");
   }
 
+  if (!(req.user._id.toString() === profile._id.toString())) {
```

Comment:
I think in this case it would make more sense to be a `PUT:/profile` with authentication, then find the profile with the decoded id

Reply:
Yes! that makes sense, I've changed it to reflect these changes.